File: benchmarks/oopsla25/subjects/062.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f062(const char *s) {
    int i = 0;
    char c0 = s[i];
    char c1 = s[i + 1];
    char c2 = s[i + 2];
    assert(strlen(s) == i + 3);
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f062(my_string);
  return 0;
}

File: benchmarks/oopsla25/subjects/062.py
#!/usr/bin/env python3

import sys

def f062(s: str):
  i = 0
  c0 = s[i]
  c1 = s[i+1]
  c2 = s[i+2]
  assert len(s) == i + 3

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f062(my_string)
